distributed/worker.py

```diff
@@ -552,13 +552,14 @@ def __init__(
             if len(protocol_address) == 2:
                 protocol = protocol_address[0]
 
-        # Target interface on which we contact the scheduler by default
-        # TODO: it is unfortunate that we special-case inproc here
-        if not host and not interface and not scheduler_addr.startswith("inproc://"):
-            host = get_ip(get_address_host(scheduler_addr))
-
         self._start_port = port
         self._start_host = host
+        if host:
+            # Helpful error message if IPv6 specified incorrectly
+            _, host_address = parse_address(host)
+            assert host_address.count(":") <= 1 or host_address.startswith(
+                "["
+            ), "Host address with IPv6 must be bracketed like '[::1]'"
```

Comment:
I've not seen `assert` used like this before. Would it be better to raise an exception here more explicitly?

Reply:
It would be different but same effect -- happy to make the change if you would like.

Other than this, does the PR look good?

Reply:
> It would be different but same effect

Functionally it would be the same but by using a more familiar pattern the code is more maintainable. (I'm sure @mrocklin wrote a good blog post about the value of writing code that a novice contributor could understand but I am failing to find it)

Otherwise yeah this seems like a great change.

Reply:
Sure, change to `ValueError` pushed.